spec/kitchen/driver/ec2_spec.rb

```diff
@@ -658,26 +683,52 @@
       include_examples "common create"
     end
 
-    context "with no key pair configured" do
+    context "and AWS SSH keys" do
       before do
-        config[:kitchen_root] = "/kitchen"
-        config.delete(:aws_ssh_key_id)
-        expect(driver).to receive(:submit_server).and_return(server)
+        allow(driver).to receive(:submit_server).and_return(server)
         allow(instance).to receive(:name).and_return("instance_name")
+      end
 
-        expect(actual_client).to receive(:create_key_pair).with(key_name: /kitchen-/).and_return(double(key_name: "kitchen-asdf", key_material: "RSA PRIVATE KEY"))
-        fake_fd = double()
-        fake_file = double()
-        allow(File).to receive(:sysopen).and_call_original
-        expect(File).to receive(:sysopen).with("/kitchen/.kitchen/instance_name.pem", kind_of(Numeric), kind_of(Numeric)).and_return(fake_fd)
-        allow(File).to receive(:open).and_call_original
-        expect(File).to receive(:open).with(fake_fd).and_yield(fake_file)
-        expect(fake_file).to receive(:write).with("RSA PRIVATE KEY")
+      context "with no AWS-managed ssh key pair configured, creates a key pair to use" do
+        before do
+          config.delete(:aws_ssh_key_id)
```

Comment:
🤔 This should probably be `config[:aws_ssh_key_id] = nil` to accurately reflect the default config where the key is present and its value is `nil`. `config.delete(:aws_ssh_key_id)` as setup is technically testing when the key is not present.